docs/docs/components/guidelines/README.md

```diff
@@ -552,4 +762,823 @@ end user. Generate and use a GUID for the shared secret value, save it to a
+
+   If the automated checks are the only symptom of a PR that's not ready for
+   review it's best to try to reach out to the author for an update (either via
+   Slack or via a comment in the PR)
+
+3. If you find a PR that you're interested in reviewing, assign yourself as the
+   reviewer to avoid other contributors from also reviewing it, as well as
+   notifying the author that their work is effectively under review.
+
+   ![How to assign yourself as a reviewer](./images/assign-pr-reviewer.gif)
+
+4. After assigning yourself as a PR reviewer, move its associated task in [the
+   Kanban board](/code/projects/1) to the **In Review** column. This helps us
+   detect any "abandoned" work sooner rather than later.
+
+   ![How to move a task to In Review status](./images/move-task-to-in-review.gif)
+
+### Guidelines
+
+When reviewing a PR, there's different aspects to consider for the review to be
+successful. Below is a list of things to pay attention to, but this list is by
+no means exhaustive and certainly does not apply to every case, so (as always)
+use your best judgment throughout the process:
+
+- [Sensible props](#sensible-props)
+- [Code structure](#code-structure)
+- [Good API usage](#good-api-usage)
+- [Robustness](#robustness)
+- [Documentation and comments](#documentation-and-comments)
+- [Styling](#styling)
+
+#### Sensible Props
+
+In the [Component Scope](#component-scope) section we mention that components
+should address specific use cases to simplify the setup process and provide a
+better user experience. However, there's cases when it makes more sense to
+provide a slightly generalized component that can be customized via props.
+
+For example, let's assume there's an event source that will emit events whenever
+a file is created in a cloud storage service. Let's name this component
+`cloud_storage-new-file-created`. For this cloud storage service it's also known
+that a common use case is to monitor for new folders, which are files of a
+specific type. Naturally, and following the guidelines above, we'd create a
+customized version of the `cloud_storage-new-file-created` component named
+`cloud_storage-new-folder-created` that reacts to newly created folders and
+nothing else.
+
+But what about other types of files? Spreadsheets and text documents are
+probably important for some users, images and CSS files for other users, and so
+on. In this case it makes much more sense to instead create a component named
+something like `cloud_storage-new-file-of-type` that is customizable via a prop
+of type `string[]` named `fileTypes` and let users select the types of files
+they prefer to watch.
+
+The advantage of making a module customizable is that it can then be extended
+into particular use cases very easily should those be highly demanded. See the
+[Code Structure](#code-structure) section for an example.
+
+There is no clear line as to which components should be customizable and which
```

Comment:
I would rephrase this section slightly to say, "If it's not clear what components you should create, reach out to the Pipedream team on the issue". We actually have a Product Manager for the team now who will be making these calls, and normally we'll have an opinion on it.

Reply:
Rephrasing the whole paragraph to this, WDYT?

> If it's unclear which components should be customizable and which ones should be
a sole component, reach out to the Pipedream team on the issue since this
usually requires input from a product perspective.